router.ts

```diff
@@ -13,19 +13,44 @@ export class Router {
         app.use(bodyParser.json());
 
         app.get('/GetShipPositions', (req, res) => {
-            let positions = myBot.getShipPositions();
-            res.send(positions);
+            try {
+                let positions = myBot.getShipPositions();
+                res.send(positions);
+            } catch(err) {
+                console.log(err);
+                res.send(err)
+            }
         });
 
         app.post('/SelectTarget', (req, res) => {
-            let target = myBot.selectTarget(req.body);
-            res.send(target);
+            try {
+                let targetPromise = myBot.selectTarget(req.body);
+                targetPromise.then((target) => {
+                    res.send(target);
+                });
+            } catch(err) {
+                console.log(err);
+                res.send(err);
+            }

```

Comment:
Nice - I like this setup for your errors. Should have made it much easier for you to debug!

(Could commonise the log & send, so you're not repeating yourself, but it's a very minor thing!